server/cluster_worker_test.go

```diff
@@ -467,23 +465,30 @@ func (s *testClusterWorkerSuite) TestHeartbeatSplit2(c *C) {
 	c.Assert(err, IsNil)
 
 	// Add Peers util all stores are used up.
-	for {
-		resp := s.heartbeatRegion(c, s.clusterID, r1, leaderPeer, false)
-		if resp == nil {
-			break
-		}
-		s.checkChangePeerRes(c, resp.GetChangePeer(), pdpb.ConfChangeType_AddNode, r1)
-	}
+	testutil.WaitUntil(c, func(c *C) bool {
+		testutil.WaitUntil(c, func(c *C) bool {
+			res := s.heartbeatRegion(c, s.clusterID, r1, leaderPeer)
+			if res == nil {
+				c.Log("no response")
+				return false
+			}
+			if res.GetChangePeer() == nil || res.GetChangePeer().GetChangeType() != pdpb.ConfChangeType_AddNode {
+				c.Log("response is not AddNode")
+				return false
+			}
+			s.onChangePeerRes(c, res.GetChangePeer(), r1)
+			return true
```

Comment:
The inner loop adds one peer each iteration, we need to add 4 peers here, so I guess add a nested level will be more clear.